drgn/commands/_builtin/crash/vm.py

```diff
@@ -0,0 +1,233 @@
+# Copyright (c) 2025, Kylin Software, Inc. and affiliates.
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import argparse
+from typing import Any, List
+
+from drgn import Object, Program
+from drgn.commands import CommandArgumentError, argument, drgn_argument
+from drgn.commands.crash import (
+    CrashDrgnCodeBuilder,
+    _pid_or_task,
+    crash_command,
+    crash_get_context,
+)
+from drgn.helpers.common.format import print_table
+from drgn.helpers.linux.fs import inode_path
+from drgn.helpers.linux.mm import for_each_vma, get_task_rss_info
+from drgn.helpers.linux.sched import task_cpu
+
+
+def _get_file_path(vma: Object, mm: Object) -> str:
+    """Get the file path name corresponding to the VMA"""
+    if vma.vm_file:
+        try:
+            path_bytes = inode_path(vma.vm_file.f_inode)
+            if path_bytes is None:
+                return "<unknown>"
+            file_path = path_bytes.decode()
+            return file_path if file_path.startswith("/") else f"/{file_path}"
+        except Exception:
+            return "<unknown>"
+
+    # Handle special memory regions
+    start = vma.vm_start.value_()
+    end = vma.vm_end.value_()
+
+    if mm.start_brk.value_() <= start and end <= mm.brk.value_():
+        return "[heap]"
+    elif start <= mm.start_stack.value_() < end:
+        return "[stack]"
+    elif (
+        hasattr(mm.context, "vdso")
+        and mm.context.vdso
+        and mm.context.vdso.value_() >= start
+        and mm.context.vdso.value_() < end
+    ):
+        return "[vdso]"
+
+    return "[anon]"
+
+
+def _get_tasks_from_args(prog: Program, pids_or_tasks: List[str]) -> List[Object]:
+    """Get task list from command line arguments"""
```

Comment:
A drgn code style quirk is that we only use docstrings for public functions (mainly because our documentation generation system uses the existence of a docstring to decide whether to include a function). Internal functions like this should just use a normal `#` comment.